src/arim/models/block_in_immersion.py

```diff
@@ -782,83 +831,58 @@ def make_paths(
     )
-
-    if max_number_of_reflection >= 1:
-        paths["LL"] = c.Path(
-            interfaces=(probe, frontwall, backwall, grid),
-            materials=(couplant_material, block_material, block_material),
-            modes=(c.Mode.L, c.Mode.L, c.Mode.L),
-            name="LL",
-        )
-
-        paths["LT"] = c.Path(
-            interfaces=(probe, frontwall, backwall, grid),
-            materials=(couplant_material, block_material, block_material),
-            modes=(c.Mode.L, c.Mode.L, c.Mode.T),
-            name="LT",
-        )
-
-        paths["TL"] = c.Path(
-            interfaces=(probe, frontwall, backwall, grid),
-            materials=(couplant_material, block_material, block_material),
-            modes=(c.Mode.L, c.Mode.T, c.Mode.L),
-            name="TL",
-        )
-
-        paths["TT"] = c.Path(
-            interfaces=(probe, frontwall, backwall, grid),
-            materials=(couplant_material, block_material, block_material),
-            modes=(c.Mode.L, c.Mode.T, c.Mode.T),
-            name="TT",
-        )
-
-    if max_number_of_reflection >= 2:
-        keys = ["LLL", "LLT", "LTL", "LTT", "TLL", "TLT", "TTL", "TTT"]
-
-        for key in keys:
-            paths[key] = c.Path(
-                interfaces=(probe, frontwall, backwall, frontwall_refl, grid),
-                materials=(
-                    couplant_material,
-                    block_material,
-                    block_material,
-                    block_material,
-                ),
-                modes=(
-                    c.Mode.L,
-                    helpers.parse_enum_constant(key[0], c.Mode),
-                    helpers.parse_enum_constant(key[1], c.Mode),
-                    helpers.parse_enum_constant(key[2], c.Mode),
-                ),
-                name=key,
+    wall_names = list(wall_dict.keys())
+
+    if (max_number_of_reflection > 0 and len(wall_names) == 0) or (
+        max_number_of_reflection > 1 and len(wall_names) < 2
+    ):
+        raise ValueError("Not enough walls to reflect from.")
+
+    mode_names = ("L", "T")
+    modes = (c.Mode.longitudinal, c.Mode.transverse)
+    for no_reflections in range(max_number_of_reflection + 1):
+        # For this number of reflections, make all the combinations of paths.
+        path_idxs_up_to_refl = list(product(range(2), repeat=no_reflections + 1))
+        for path_idxs in path_idxs_up_to_refl:
+            # For each path with this number of reflections.
+            path_name = ""  # Current convention does not include frontwall transmission in path name, so start with empty string.
+            path_modes = [c.Mode.longitudinal]
+            path_interfaces = [probe, frontwall]
+            path_materials = [couplant_material]
+
+            for i, mode in enumerate(path_idxs):
+                if i == 0:
+                    path_name += mode_names[mode]
+                else:
+                    # Have to settle on a naming convention.
```

Comment:
I never imagined that naming paths and views would become so complex! It's probably best to aim for unambiguous names (using `longname` instead of `name`) as dict keys; the end user is still free to use the short name as a plot title etc. if they wish.